papyrus/geo_interface.py

```diff
@@ -100,10 +109,14 @@ def __update__(self, feature):
                     self._shape = shape
             elif not col.primary_key:
                 if p.key in feature.properties:
+                    if col.info.get('readonly'):
+                        continue
                     setattr(self, p.key, feature.properties[p.key])
 
         if self.__add_properties__:
                 for k in self.__add_properties__:
+                    if col.info.get('readonly'):
```

Comment:
`col` is out of scope here.

Reply:
As we do not know the types of added properties, I would propose to not support 'readonly' here.